lib/src/view/game/game_result_dialog.dart

```diff
@@ -97,7 +101,34 @@ class _GameResultDialogState extends ConsumerState<GameResultDialog> {
                             gameState.game.opponent?.onGame == true &&
                             gameState.game.opponent?.offeringRematch != true
                         ? () {
-                            ref.read(ctrlProvider.notifier).proposeOrAcceptRematch();
+                            final game = gameState.game;
+
+                            // Check if it was an Odds Bot game
+                            if (game.meta.variant == Variant.fromPosition) {
+                              // If yes, build the detailed challenge and navigate to the new GameScreen
+                              Navigator.of(context, rootNavigator: true).push(
+                                GameScreen.buildRoute(
+                                  context,
+                                  challenge: ChallengeRequest(
+                                    destUser: game.opponent!.user!,
+                                    variant: Variant.fromPosition,
+                                    timeControl: ChallengeTimeControlType.clock,
+                                    clock: (
+                                      time: game.meta.clock!.initial,
+                                      increment: game.meta.clock!.increment,
+                                    ),
+                                    rated: false,
```

Comment:
In theory you can create a rated from position challenge, so you should use the game information to determine if the challenge is rated or not.